standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ServletSecurity.java

```diff
@@ -310,7 +310,18 @@ static void loginServerPrincipal(Configuration conf) throws IOException {
    * @throws IOException if getting password fails
    */
   static SslContextFactory createSslContextFactory(Configuration conf) throws IOException {
-    final boolean useSsl  = MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.USE_SSL);
+    return createSslContextFactory(conf, MetastoreConf.ConfVars.USE_SSL);
+  }
+
+  /**
+   * Creates an SSL context factory if configuration states so.
+   * @param conf the configuration
+   * @param condition the condition variable to check for SSL
+   * @return null if no ssl in config, an instance otherwise
+   * @throws IOException if getting password fails
+   */
+  static SslContextFactory createSslContextFactory(Configuration conf, MetastoreConf.ConfVars condition) throws IOException {
```

Comment:
nit: the method name suggest that this method creates a SSL Context Factory. But it is conditional based on whether or not a config is enabled. If you do not have a better name, its fine as well.